redfish/internal/controller/http/v1/handler/middleware.go

```diff
@@ -0,0 +1,54 @@
+// Package v1 provides Redfish v1 API middleware.
+package v1
+
+import (
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/device-management-toolkit/console/config"
+)
+
+// RedfishJWTAuthMiddleware returns a Gin middleware that validates JWT tokens
+// and returns Redfish-compliant error responses for authentication failures.
+func RedfishJWTAuthMiddleware(jwtKey string, verifier *oidc.IDTokenVerifier) gin.HandlerFunc {
```

Comment:
I would expect this to use what we currently have for jwtauth middlewater, having two different paths for authentication would be very confusing from a developer experience perspective.

Reply:
HTTP Basic authentication is a requirement coming from redfish https://redfish.dmtf.org/schemas/DSP0266_1.15.1.html#authentication